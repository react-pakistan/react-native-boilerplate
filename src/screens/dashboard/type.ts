export interface IProject {
  banner : string;
  description : string;
  id : string;
  label : string;
}
