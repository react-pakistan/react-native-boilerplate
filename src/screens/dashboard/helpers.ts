import { IProject } from './type';

export const dashboardScreenText = {
  mainHeading: 'React Pakistan Open Source Projects',
};

export const openSourceMainProjects : Array<IProject> = [
  {
    banner: 'https://res.cloudinary.com/dq6hflqwx/image/upload/v1565548545/GitHub/React_Pakistan_Storybook_Logo.jpg',
    description: `React Pakistan is an OpenSource & Free Software (FOSS) platform,
      creating React software solutions for the global community.`.replace(/[\n\r]+ */g, ' '),
    id: 'react-pakistan-open-source-main-project-1',
    label: 'React Pakistan',
  },
];

export const openSourceMobileProjects : Array<IProject> = [
  {
    banner: 'https://res.cloudinary.com/dq6hflqwx/image/upload/v1548862588/react-native-boilerplate.jpg',
    description: `A collection of curated and well maintained React Native
      boilerplates with various possibilities and combinations.`.replace(/[\n\r]+ */g, ' '),
    id: 'react-pakistan-open-source-mobile-project-1',
    label: 'React Native Boilerplates',
  },
  {
    banner: 'https://res.cloudinary.com/dq6hflqwx/image/upload/v1583650645/GitHub/react-native-commons-collection.jpg',
    description: `A generic, Commons library for React Native apps, following modern approach by building your
      project's UI in isolation using state of the art Storybook along with React Native.`.replace(/[\n\r]+ */g, ' '),
    id: 'react-pakistan-open-source-mobile-project-2',
    label: 'React Native Commons Collection',
  },
  {
    banner: 'https://res.cloudinary.com/dq6hflqwx/image/upload/v1583650649/GitHub/react-native-ui-collection.jpg',
    description: `A generic, UI library for React Native apps, following modern approach by building your
      project's UI in isolation using state of the art Storybook along with React Native.`.replace(/[\n\r]+ */g, ' '),
    id: 'react-pakistan-open-source-mobile-project-3',
    label: 'React Native UI Collection',
  },
  {
    banner: 'https://res.cloudinary.com/dq6hflqwx/image/upload/v1583650647/GitHub/react-native-icon-collection.jpg',
    description: `A generic, Icon library for React Native apps, following modern approach by building your
      project's UI in isolation using state of the art Storybook along with React Native.`.replace(/[\n\r]+ */g, ' '),
    id: 'react-pakistan-open-source-mobile-project-4',
    label: 'React Native Icon Collection',
  },
  {
    banner: 'https://res.cloudinary.com/dq6hflqwx/image/upload/v1583650647/GitHub/react-native-icon-collection.jpg',
    description: `A generic, Logo library for React Native apps, following modern approach by building your
      project's UI in isolation using state of the art Storybook along with React Native.`.replace(/[\n\r]+ */g, ' '),
    id: 'react-pakistan-open-source-mobile-project-5',
    label: 'React Native Logo Collection',
  },
  {
    banner: 'https://res.cloudinary.com/dq6hflqwx/image/upload/v1583650647/GitHub/react-native-icon-collection.jpg',
    description: `A generic, Logo library for React Native apps, following modern approach by building your
      project's UI in isolation using state of the art Storybook along with React Native.`.replace(/[\n\r]+ */g, ' '),
    id: 'react-pakistan-open-source-mobile-project-6',
    label: 'React Native Emoji Collection',
  },
];

export const openSourceWebProjects : Array<IProject> = [
  {
    banner: 'https://res.cloudinary.com/dq6hflqwx/image/upload/v1566355017/GitHub/react-commons-collection.jpg',
    description: `A generic, Commons library for React web apps, following modern approach by building your
      project's UI in isolation using state of the art Storybook along with React.`.replace(/[\n\r]+ */g, ' '),
    id: 'react-pakistan-open-source-web-project-1',
    label: 'React Commons Collection',
  },
  {
    banner: 'https://res.cloudinary.com/dq6hflqwx/image/upload/v1566355010/GitHub/react-ui-collection.jpg',
    description: `A generic, UI library for React web apps, following modern approach by building your
      project's UI in isolation using state of the art Storybook along with React.`.replace(/[\n\r]+ */g, ' '),
    id: 'react-pakistan-open-source-web-project-2',
    label: 'React UI Collection',
  },
  {
    banner: 'https://res.cloudinary.com/dq6hflqwx/image/upload/v1567499955/GitHub/react-icons-collection.jpg',
    description: `A generic, Icon library for React web apps, following modern approach by building your
      project's UI in isolation using state of the art Storybook along with React.`.replace(/[\n\r]+ */g, ' '),
    id: 'react-pakistan-open-source-web-project-3',
    label: 'React Icon Collection',
  },
  {
    banner: 'https://res.cloudinary.com/dq6hflqwx/image/upload/v1567499957/GitHub/react-logo-collection.jpg',
    description: `A generic, Logo library for React web apps, following modern approach by building your
      project's UI in isolation using state of the art Storybook along with React.`.replace(/[\n\r]+ */g, ' '),
    id: 'react-pakistan-open-source-web-project-4',
    label: 'React Logo Collection',
  },
  {
    banner: 'https://res.cloudinary.com/dq6hflqwx/image/upload/v1567499957/GitHub/react-emoji-collection.jpg',
    description: `A generic, Logo library for React web apps, following modern approach by building your
      project's UI in isolation using state of the art Storybook along with React.`.replace(/[\n\r]+ */g, ' '),
    id: 'react-pakistan-open-source-web-project-5',
    label: 'React Emoji Collection',
  },
  {
    banner: 'https://res.cloudinary.com/dq6hflqwx/image/upload/v1567499950/GitHub/react-design-collection.jpg',
    description: `A generic, Design library for React web apps, following modern approach by building your
      project's UI in isolation using state of the art Storybook along with React.`.replace(/[\n\r]+ */g, ' '),
    id: 'react-pakistan-open-source-web-project-6',
    label: 'React Design Collection',
  },
];
