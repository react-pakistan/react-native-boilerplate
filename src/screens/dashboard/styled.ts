import { Text, View } from '@react-pakistan/react-native-commons-collection';
import { IRNTheme } from '@react-pakistan/util-react-native-functions';
import { ThemedStyledProps } from 'styled-components';
import styled from 'styled-components/native';

export const DashboardWrapper = styled(View)`
  align-items: center;
  justify-content: center;
`;

export const DashboardHeading = styled(Text)``;

export const ListHeadingWrapper = styled(View)`
  align-items: center;
  justify-content: center;
  padding-horizontal: ${({
    theme,
  } : ThemedStyledProps<unknown, IRNTheme>) : number => theme.spacing.default}px;
  padding-vertical: ${({
    theme,
  } : ThemedStyledProps<unknown, IRNTheme>) : number => theme.spacing.default}px;
`;
