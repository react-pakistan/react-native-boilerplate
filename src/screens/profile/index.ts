export * from './profile';
export * from './type';
