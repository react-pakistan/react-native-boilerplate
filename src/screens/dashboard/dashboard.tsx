import React, { ReactElement } from 'react';
import { FlatList, ListRenderItem } from 'react-native';
import { useTheme } from 'styled-components/native';
import { BannerItem } from '../../components/banner-item';
import { dashboardScreenText, openSourceMainProjects } from './helpers';
import {
  DashboardHeading,
  DashboardWrapper,
  ListHeadingWrapper,
} from './styled';
import { IProject } from './type';

const renderItem : ListRenderItem<IProject> = (
  {
    item: {
      banner,
      description,
      label,
    },
  },
) : ReactElement => (
  <BannerItem
    banner={banner}
    description={description}
    label={label}
  />
);

export const Dashboard = () : ReactElement => {
  // theme
  const theme = useTheme();

  return (
    <DashboardWrapper>
      <FlatList
        data={openSourceMainProjects}
        keyExtractor={({ id } : IProject) : string => id}
        ListHeaderComponent={(
          <ListHeadingWrapper>
            <DashboardHeading
              {...theme.typography.h1}
            >
              {dashboardScreenText.mainHeading}
            </DashboardHeading>
          </ListHeadingWrapper>
        )}
        renderItem={renderItem}
      />
    </DashboardWrapper>
  );
};
