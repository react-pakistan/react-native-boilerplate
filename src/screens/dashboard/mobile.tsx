import { IRNTheme } from '@react-pakistan/util-react-native-functions';
import React, { ReactElement, memo } from 'react';
import { FlatList, ListRenderItem } from 'react-native';
import { withTheme } from 'styled-components/native';
import { BannerItem } from '../../components/banner-item';
import { dashboardScreenText, openSourceMobileProjects } from './helpers';
import {
  DashboardHeading,
  DashboardWrapper,
  ListHeadingWrapper,
} from './styled';
import { IProject } from './type';

export const DashboardMobile = memo(withTheme(({
  theme,
} : IDashboardMobileProps) : ReactElement => {
  const renderItem : ListRenderItem<IProject> = ({ item: { banner, description, label } }) : ReactElement => (
    <BannerItem
      banner={banner}
      description={description}
      label={label}
    />
  );

  return (
    <DashboardWrapper>
      <FlatList
        data={openSourceMobileProjects}
        keyExtractor={({ id } : IProject) : string => id}
        ListHeaderComponent={(
          <ListHeadingWrapper>
            <DashboardHeading
              {...theme.typography.h1}
            >
              {dashboardScreenText.mainHeading}
            </DashboardHeading>
          </ListHeadingWrapper>
        )}
        renderItem={renderItem}
      />
    </DashboardWrapper>
  );
}));

export interface IDashboardMobileProps {
  /**
   *
   */
  theme : IRNTheme;
}
