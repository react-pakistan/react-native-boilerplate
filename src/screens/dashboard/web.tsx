import { IRNTheme } from '@react-pakistan/util-react-native-functions';
import React, { ReactElement, memo } from 'react';
import { FlatList, ListRenderItem } from 'react-native';
import { withTheme } from 'styled-components/native';
import { BannerItem } from '../../components/banner-item';
import { dashboardScreenText, openSourceWebProjects } from './helpers';
import {
  DashboardHeading,
  DashboardWrapper,
  ListHeadingWrapper,
} from './styled';
import { IProject } from './type';

export const DashboardWeb = memo(withTheme(({
  theme,
} : IDashboardWebProps) : ReactElement => {
  const renderItem : ListRenderItem<IProject> = ({ item: { banner, description, label } }) : ReactElement => (
    <BannerItem
      banner={banner}
      description={description}
      label={label}
    />
  );

  return (
    <DashboardWrapper>
      <FlatList
        data={openSourceWebProjects}
        keyExtractor={({ id } : IProject) : string => id}
        ListHeaderComponent={(
          <ListHeadingWrapper>
            <DashboardHeading
              {...theme.typography.h1}
            >
              {dashboardScreenText.mainHeading}
            </DashboardHeading>
          </ListHeadingWrapper>
        )}
        renderItem={renderItem}
      />
    </DashboardWrapper>
  );
}));

export interface IDashboardWebProps {
  /**
   *
   */
  theme : IRNTheme;
}
