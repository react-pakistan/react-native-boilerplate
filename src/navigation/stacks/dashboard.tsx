import { IEmptyAction } from '@react-pakistan/util-functions/action-creator-factories';
import { DrawerActions } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import { Icon } from '@react-pakistan/react-native-icon-collection/icon';
import React, { ReactElement, ReactNode } from 'react';
import { useDispatch } from 'react-redux';
import { hideAppAction } from '../../redux/actions';
import { TYPES } from '../../redux/constants';
import { Dashboard as DashboardScreen } from '../../screens/dashboard';
import { NAVIGATION_ROUTES, NAVIGATION_TITLES } from '../navigation-routes';
import { HeaderIcon } from '../styled';

const Stack = createStackNavigator();

export const DashboardStack = () : ReactElement => {
  // dispatch
  const dispatch = useDispatch();

  return (
    <Stack.Navigator
      screenOptions={{
        headerShown: false,
      }}
    >
      <Stack.Screen
        component={DashboardScreen}
        name={NAVIGATION_ROUTES.TAB_MAIN}
        options={{
          headerRight: () : ReactNode => (
            <HeaderIcon
              onPress={() : IEmptyAction<TYPES.HIDE_APP> => dispatch(hideAppAction())}
            >
              <Icon
                icon='MobileUiGrey9'
              />
            </HeaderIcon>
          ),
          headerLeft: () : ReactNode => (
            <HeaderIcon
              onPress={() : any => DrawerActions.toggleDrawer()}
            >
              <Icon
                icon='MobileUiGrey15'
              />
            </HeaderIcon>
          ),
          headerTitle: NAVIGATION_TITLES.TAB_MAIN,
        }}
      />
    </Stack.Navigator>
  );
};
