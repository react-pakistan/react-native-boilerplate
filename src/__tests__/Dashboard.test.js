import { ThemeProvider } from 'styled-components/native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { Dashboard } from '../screens/dashboard';
import { appTheme } from '../theme';

test('renders Dashboard component correctly', () => {
  const renderer = new ShallowRenderer();
  renderer.render(
    <ThemeProvider theme={appTheme}>
      <Dashboard />
    </ThemeProvider>
  );
  const wrapper = renderer.getRenderOutput();
  expect(wrapper).toMatchSnapshot();
});
