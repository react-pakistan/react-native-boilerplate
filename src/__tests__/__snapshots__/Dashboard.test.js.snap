// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Dashboard component correctly 1`] = `
<Context.Provider
  value={
    Object {
      "breakpoints": Object {
        "desktop": "(min-width: 2560px)",
        "laptop": "(min-width: 1024px)",
        "laptopL": "(min-width: 1440px)",
        "mobileL": "(min-width: 425px)",
        "mobileM": "(min-width: 375px)",
        "mobileS": "(min-width: 320px)",
        "tablet": "(min-width: 768px)",
      },
      "colors": Object {
        "black": "#2E2E2E",
        "cherryRed": "#F62F5E",
        "darkGrey": "#EFEFEF",
        "inputBorder": "#E1E1E1",
        "lightBlack": "#6C6C6C",
        "lightGrey": "#F7F7F7",
        "lighterBlack": "#B4B4B4",
        "limeYellow": "#EFFC90",
        "paleGreen": "#00D3CA",
        "rustyGold": "#F1AD3D",
        "skyBlue": "#1E90FF",
        "white": "#FFFFFF",
      },
      "misc": Object {
        "darkMode": false,
      },
      "radius": Object {
        "default": 8,
        "double": 16,
        "full": 100,
        "half": 60,
      },
      "shadow": Object {
        "default": "0px 1px 4px #E5E9F2",
        "medium": "-4px 0 12px rgba(0, 0, 0, 0.3)",
      },
      "spacing": Object {
        "default": 32,
        "huge": 96,
        "large": 48,
        "medium": 24,
        "small": 16,
        "xlarge": 64,
        "xsmall": 8,
      },
      "typography": Object {
        "fontFamily": "Roboto-Regular",
        "h1": Object {
          "color": "#2E2E2E",
          "fontFamily": "Roboto-Bold",
          "fontSize": 56,
          "letterSpacing": 0,
          "lineHeight": 56,
          "textAlign": "left",
          "textTransform": "uppercase",
        },
        "h2": Object {
          "color": "#2E2E2E",
          "fontFamily": "Roboto-Medium",
          "fontSize": 44,
          "letterSpacing": 0,
          "lineHeight": 56,
          "textAlign": "left",
          "textTransform": "none",
        },
        "h3": Object {
          "color": "#2E2E2E",
          "fontFamily": "Roboto-Regular",
          "fontSize": 32,
          "letterSpacing": 0,
          "lineHeight": 56,
          "textAlign": "left",
          "textTransform": "none",
        },
        "h4": Object {
          "fontFamily": "Roboto-Medium",
          "fontSize": 40,
          "letterSpacing": 0,
          "lineHeight": 44,
          "textAlign": "left",
          "textTransform": "none",
        },
        "h5": Object {
          "fontFamily": "Roboto-Medium",
          "fontSize": 36,
          "letterSpacing": 0,
          "lineHeight": 40,
          "textAlign": "left",
          "textTransform": "none",
        },
        "h6": Object {
          "fontFamily": "Roboto-Medium",
          "fontSize": 32,
          "letterSpacing": 0,
          "lineHeight": 36,
          "textAlign": "left",
          "textTransform": "none",
        },
        "p1": Object {
          "fontFamily": "Roboto-Regular",
          "fontSize": 56,
          "letterSpacing": 0,
          "lineHeight": 60,
          "textAlign": "left",
          "textTransform": "none",
        },
        "p2": Object {
          "fontFamily": "Roboto-Regular",
          "fontSize": 52,
          "letterSpacing": 0,
          "lineHeight": 56,
          "textAlign": "left",
          "textTransform": "none",
        },
        "p3": Object {
          "fontFamily": "Roboto-Regular",
          "fontSize": 48,
          "letterSpacing": 0,
          "lineHeight": 52,
          "textAlign": "left",
          "textTransform": "none",
        },
        "p4": Object {
          "fontFamily": "Roboto-Regular",
          "fontSize": 40,
          "letterSpacing": 0,
          "lineHeight": 44,
          "textAlign": "left",
          "textTransform": "none",
        },
        "p5": Object {
          "fontFamily": "Roboto-Regular",
          "fontSize": 36,
          "letterSpacing": 0,
          "lineHeight": 40,
          "textAlign": "left",
          "textTransform": "none",
        },
        "p6": Object {
          "fontFamily": "Roboto-Regular",
          "fontSize": 32,
          "letterSpacing": 0,
          "lineHeight": 36,
          "textAlign": "left",
          "textTransform": "none",
        },
        "text": Object {
          "color": "#2E2E2E",
          "fontFamily": "Roboto-Regular",
          "fontSize": 28,
          "letterSpacing": 0,
          "lineHeight": 40,
          "textAlign": "left",
          "textTransform": "none",
        },
      },
      "zIndex": Object {
        "dropdown": 100,
        "modal": 300,
        "pageFooter": 200,
      },
    }
  }
>
  <Dashboard />
</Context.Provider>
`;
